launch/README.md

```diff
@@ -0,0 +1,41 @@
+  * `aws`: Uses Amazon Web Services could formation console. Supports both zen and simple templates. (Can only be used with `platform: aws`. Requires: `template_url`, `template_parameters`
+  * `azure`: Uses Azure Resource Manager deployment templates. Supports both ACS and DC/OS templates. (Can only be used with `platform: azure`. Requires `template_url`, and `template_parameters`
+  * `onprem`: Uses the DC/OS bash installer to orchestra a deployment on arbitrary hosts of a bare cluster. Requires `num_masters`, `num_private_agents`, `num_public_agents`, `installer_url`, `instance_type`, `os_name`, and `dcos_config`
+
+### Conditionally Required Fields
+* `ssh_user`: If `provider: onprem` is used, then the host VM configuration is known to dcos-launch and this value will be calculated. Otherwise, it should always be supplied, and must be supplied for `provider: onprem`
+* `ssh_private_key_filename`: If `key_helper` is `true` then this field cannot be supplied. Otherwise it should always be specified, and it is absolutely requred for `onprem` deploy
+* `instance_count`: If `deploy_bare_cluster_only: true` (see below), then this field must be provided. Note: when this bare cluster is used again with dcos-launch, the instance count must match `1 + num_master + num_private_agents + num_public_agents`
+* `aws_key_name`: If `key_helper: false` and `provider: onprem` and `platform: aws`, then a pre-existing EC2 SSH KeyPair must be supplied for launching the VPC
+_Note_: DC/OS deployed from aws or azure provider do not technically need `ssh_user` or `ssh_private_key_filename`. However, without this additional data, the integration tests will not be trigger-able from dcos-launch. Thus, it is not recommended, but allowable, to omit these fields when not using the onprem provider
+
+### Options
+* `key_helper`: generate private SSH keys for the underlying hosts if `true`. In `platform: aws`, this means the user does not have to supply `KeyName` in the template parameters and dcos-launch will fill it in. Similarly, in `platform: azure`, `sshRSAPublicKey` is populated automatically. In the aws case, this key will be deleted from EC2 when the deployment is deleted with dcos-launch
+* `zen_helper`: only to be used with `provider: aws` and zen templates. If `true` the network prerequisites for launching a zen cluster will be provided if missing. As with `key_helper`, these resources will be deleted if dcos-launch is used for destroying the deployment
+* `deploy_bare_cluster_only`: only to be used with `provider: onprem`. If `true`, the onprem deployment will stop after the bare cluster hosts have been provisioned. Like any dcos-launch deployment, this will generate a cluster info JSON. This JSON can then be passed back into another dcos-launch deployment for `provider: onprem` with `platform: bare_cluster`, by providing the field `platform_info_filename`
```

Comment:
Eh, it's not a variable name and it's not proper English, so I will compromise with `cluster-info`